{"nbformat":4,"nbformat_minor":5,"metadata":{"notebookId":"3cc8788b-f7e2-4163-b70e-20d7f9cbdc32","language_info":{"pygments_lexer":"ipython3","file_extension":".py","version":"3.7.7","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"name":"python","nbconvert_exporter":"python"},"kernelspec":{"name":"python3","display_name":"Yandex DataSphere Kernel","language":"python"}},"cells":[{"cell_type":"code","source":"from hidt_model import HiDTModel","metadata":{"cellId":"fqu0g9kon8hdtjz25b359l"},"outputs":[],"execution_count":null},{"cell_type":"code","source":"def main():\n    ","metadata":{"cellId":"irrhqjktnhfpkp8zfqmctg"},"outputs":[],"execution_count":null},{"cell_type":"code","source":"#!:bash\n!git rebase origin/dev","metadata":{"cellId":"n4fnwloch7jz8vl0v367si","trusted":true},"outputs":[{"output_type":"stream","name":"stdout","text":"First, rewinding head to replay your work on top of it...\nApplying: init\nUsing index info to reconstruct a base tree...\n.git/rebase-apply/patch:40: trailing whitespace.\n    \"lambdas\": [5, 2, 1, 0.1, 4, 1] \n.git/rebase-apply/patch:1205: trailing whitespace.\n        \n.git/rebase-apply/patch:1253: trailing whitespace.\n    def __init__(self, \n.git/rebase-apply/patch:1266: trailing whitespace.\n        self.history = {\"train\": defaultdict(list), \n.git/rebase-apply/patch:1288: trailing whitespace.\n    def load_checkpoint(self, \nwarning: squelched 6 whitespace errors\nwarning: 11 lines add whitespace errors.\nFalling back to patching base and 3-way merge...\nAuto-merging README.md\nCONFLICT (add/add): Merge conflict in README.md\nerror: Failed to merge in the changes.\nPatch failed at 0001 init\nUse 'git am --show-current-patch' to see the failed patch\n\nResolve all conflicts manually, mark them as resolved with\n\"git add/rm <conflicted_files>\", then run \"git rebase --continue\".\nYou can instead skip this commit: run \"git rebase --skip\".\nTo abort and get back to the state before \"git rebase\", run \"git rebase --abort\".\n\n"},{"output_type":"error","ename":"Exception","evalue":"Process exited with code 128","traceback":["\u001B[0;31m---------------------------------------------------------------------------\u001B[0m","\u001B[0;31mException\u001B[0m                                 Traceback (most recent call last)","\u001B[0;32m<ipython-input-1-86a64c445d85>\u001B[0m in \u001B[0;36m<module>\u001B[0;34m\u001B[0m\n\u001B[0;32m----> 1\u001B[0;31m \u001B[0mexecute_script\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'bash'\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0;34m'\\ngit rebase origin/dev\\n#'\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m","\u001B[0;32m/kernel/lib/python3.7/site-packages/ml_kernel/script_executor.py\u001B[0m in \u001B[0;36mexecute\u001B[0;34m(self, lang, code)\u001B[0m\n\u001B[1;32m     33\u001B[0m         \u001B[0mreturn_code\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_system\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mcode\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m     34\u001B[0m         \u001B[0;32mif\u001B[0m \u001B[0mreturn_code\u001B[0m \u001B[0;34m!=\u001B[0m \u001B[0;36m0\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m---> 35\u001B[0;31m             \u001B[0;32mraise\u001B[0m \u001B[0mException\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'Process exited with code %d'\u001B[0m \u001B[0;34m%\u001B[0m \u001B[0mreturn_code\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m","\u001B[0;31mException\u001B[0m: Process exited with code 128"]}],"execution_count":19},{"cell_type":"code","source":"!git add .","metadata":{"cellId":"hwkkajl4knjp292wahsbyd","trusted":true},"outputs":[],"execution_count":20},{"cell_type":"code","source":"!git rebase --continue","metadata":{"cellId":"x4hzhzvwglssa5q3sczfnn","trusted":true},"outputs":[{"output_type":"stream","name":"stdout","text":"Applying: init\nApplying: merge\nUsing index info to reconstruct a base tree...\nM\tREADME.md\nFalling back to patching base and 3-way merge...\nAuto-merging README.md\nCONFLICT (content): Merge conflict in README.md\nerror: Failed to merge in the changes.\nPatch failed at 0002 merge\nUse 'git am --show-current-patch' to see the failed patch\n\nResolve all conflicts manually, mark them as resolved with\n\"git add/rm <conflicted_files>\", then run \"git rebase --continue\".\nYou can instead skip this commit: run \"git rebase --skip\".\nTo abort and get back to the state before \"git rebase\", run \"git rebase --abort\".\n\n"},{"output_type":"error","ename":"Exception","evalue":"Process exited with code 128","traceback":["\u001B[0;31m---------------------------------------------------------------------------\u001B[0m","\u001B[0;31mException\u001B[0m                                 Traceback (most recent call last)","\u001B[0;32m<ipython-input-3-15ca9da11634>\u001B[0m in \u001B[0;36m<module>\u001B[0;34m\u001B[0m\n\u001B[0;32m----> 1\u001B[0;31m \u001B[0mget_ipython\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msystem\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'git rebase --continue'\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m      2\u001B[0m \u001B[0;31m#\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n","\u001B[0;32m/kernel/lib/python3.7/site-packages/ml_kernel/kernel.py\u001B[0m in \u001B[0;36m<lambda>\u001B[0;34m(code)\u001B[0m\n\u001B[1;32m    267\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    268\u001B[0m         \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_script_executor\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0mScriptExecutor\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mProcessHandler\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msystem\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m--> 269\u001B[0;31m         \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mshell\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msystem\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0;32mlambda\u001B[0m \u001B[0mcode\u001B[0m\u001B[0;34m:\u001B[0m \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_script_executor\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mexecute\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m\"bash\"\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mcode\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m    270\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    271\u001B[0m     \u001B[0;32mdef\u001B[0m \u001B[0m_init_message_handlers\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mself\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n","\u001B[0;32m/kernel/lib/python3.7/site-packages/ml_kernel/script_executor.py\u001B[0m in \u001B[0;36mexecute\u001B[0;34m(self, lang, code)\u001B[0m\n\u001B[1;32m     33\u001B[0m         \u001B[0mreturn_code\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_system\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mcode\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m     34\u001B[0m         \u001B[0;32mif\u001B[0m \u001B[0mreturn_code\u001B[0m \u001B[0;34m!=\u001B[0m \u001B[0;36m0\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m---> 35\u001B[0;31m             \u001B[0;32mraise\u001B[0m \u001B[0mException\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'Process exited with code %d'\u001B[0m \u001B[0;34m%\u001B[0m \u001B[0mreturn_code\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m","\u001B[0;31mException\u001B[0m: Process exited with code 128"]}],"execution_count":21},{"cell_type":"code","source":"","metadata":{"cellId":"yjjh996n80mn0vaz7yrfl9"},"outputs":[],"execution_count":null}]}